graphqlapi/src/main/java/io/stargate/graphql/schema/schemafirst/fetchers/dynamic/DeleteFetcher.java

```diff
@@ -54,36 +62,36 @@ public DeleteFetcher(
   }
 
   @Override
-  protected Boolean get(
+  protected Object get(
       DataFetchingEnvironment environment,
       DataStore dataStore,
       AuthenticationSubject authenticationSubject)
       throws UnauthorizedException {
 
     EntityModel entityModel = model.getEntity();
     Keyspace keyspace = dataStore.schema().keyspace(entityModel.getKeyspaceName());
-    Map<String, Object> input = environment.getArgument(model.getEntityArgumentName());
-    Collection<BuiltCondition> conditions = new ArrayList<>();
-    for (FieldModel column : entityModel.getPrimaryKey()) {
-      String graphqlName = column.getGraphqlName();
-      Object graphqlValue = input.get(graphqlName);
-      if (graphqlValue == null) {
-        throw new IllegalArgumentException("Missing value for field " + graphqlName);
-      } else {
-        conditions.add(
-            BuiltCondition.of(
-                column.getCqlName(),
-                Predicate.EQ,
-                toCqlValue(graphqlValue, column.getCqlType(), keyspace)));
-      }
+
+    // We're either getting the values from a single entity argument, or individual PK field
+    // arguments:
+    Function<String, Object> fieldGetter;
+    if (model.getEntityArgumentName().isPresent()) {
+      Map<String, Object> entity = environment.getArgument(model.getEntityArgumentName().get());
+      fieldGetter = entity::get;
+    } else {
+      fieldGetter = environment::getArgument;
     }
 
+    Collection<BuiltCondition> conditions = new ArrayList<>();
+    bindPartitionKey(fieldGetter, keyspace, conditions);
```

Comment:
those two methods (`bindPartitionKey` and `bindClusteringColumns`) seem like a good candidate for reusing in other Fetchers. Is it possible?

Reply:
Yes, I plan to revisit and unify condition handling, I'll clean that up.